### PR TITLE
Window minimize crash fix: #37327

### LINKED ISSUE #37327: Editor crash when minimizing on Windows (crash in VulkanContext::prepare_buffers)
**Godot version:**
5f1107a

**OS/device including version:**
Windows 10

**Steps to reproduce:**
1. Minimize editor
```
[0] vk_optimusGetInstanceProcAddr
[1] vkGetInstanceProcAddr
[2] VulkanContext::prepare_buffers (x:\inne\godot\source\godot\drivers\vulkan\vulkan_context.cpp:1228)
[3] RenderingDeviceVulkan::prepare_screen_for_drawing (x:\inne\godot\source\godot\drivers\vulkan\rendering_device_vulkan.cpp:6692)
[4] VisualServerViewport::draw_viewports (x:\inne\godot\source\godot\servers\visual\visual_server_viewport.cpp:452)
[5] VisualServerRaster::draw (x:\inne\godot\source\godot\servers\visual\visual_server_raster.cpp:114)
[6] Main::iteration (x:\inne\godot\source\godot\main\main.cpp:2139)
[7] OS_Windows::run (x:\inne\godot\source\godot\platform\windows\os_windows.cpp:661)
[8] widechar_main (x:\inne\godot\source\godot\platform\windows\godot_windows.cpp:162)
[9] _main (x:\inne\godot\source\godot\platform\windows\godot_windows.cpp:186)
[10] main (x:\inne\godot\source\godot\platform\windows\godot_windows.cpp:196)
[11] __scrt_common_main_seh (f:\dd\vctools\crt\vcstartup\src\startup\exe_common.inl:283)
[12] BaseThreadInitThunk
-- END OF BACKTRACE --
```

### PR DESCRIPTION
**Pull request by ThakeeNathees**
Fix: #37327

### CODE REVIEW

File: platform/windows/display_server_windows.cpp [hunk 1]
```diff
@@ -1093,6 +1093,7 @@ bool DisplayServerWindows::can_any_window_draw() const {
 	_THREAD_SAFE_METHOD_
 
 	for (Map<WindowID, WindowData>::Element *E = windows.front(); E; E = E->next()) {
+		if (E->key() == -1) continue;
```
Comment (bruvzg) [inline platform/windows/display_server_windows.cpp]: Should be `INVALID_WINDOW_ID` (which is `-1`) instead of `-1`.

Comment (bruvzg) [reply platform/windows/display_server_windows.cpp]: Some missing checks I found:

https://github.com/godotengine/godot/blob/d2d359d67c5c35351ed953448430536c57db1522/platform/windows/display_server_windows.cpp#L498-L500

`_create_window` can return -1

https://github.com/godotengine/godot/blob/d2d359d67c5c35351ed953448430536c57db1522/platform/windows/display_server_windows.cpp#L1788-L1795

Can potentially use -1 if window is not in the map (window is created but not added to the map, or removed from the map but still closing, not sure if any of this is possible).

**Comment by reduz** [comment]
Fix does not look correct, I have the feeling the problem is the creation flags for the minimized window..

**Comment by ThakeeNathees** [comment]
after the @bruvzg 's review I think that this is not a proper fix. I've just implemented some error fails for the crash and it worked. should I close this?
@reduz

**Comment by HaSa1002** [comment]
You should catch it in the WndProc function with
```
if (window_id == INVALID_WINDOW_ID)
    return;
```
https://github.com/godotengine/godot/blob/d29514accef09ba0402cdbb87ec5f5e773a48bbc/platform/windows/display_server_windows.cpp#L1806

**Comment by marstaik** [comment]
@reduz @HaSa1002  I actually think this PR is correct. There are no windows creation flags for Vulkan, seen here: https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VkWin32SurfaceCreateFlagsKHR.html

Also, according to here: https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VkWin32SurfaceCreateFlagsKHR

>Note
>
> Due to above restrictions, it is only possible to create a new swapchain on this platform with imageExtent being equal to the current size of the window.
>
> The window size may become (0, 0) on this platform (e.g. when the window is minimized), and so a swapchain cannot be created until the size changes.

which the code does, it sees windows size being 0,0 and returns. But there is no code for handling a nullptr swapchain further up.

**Comment by KoBeWi** [comment]
The original issue is already fixed. Thanks for the contribution nonetheless.

**Pull request closed without merging**